src/transaction.ts

```diff
@@ -81,6 +80,31 @@ enum TransactionState {
   IN_PROGRESS, // IN_PROGRESS currently tracks the expired state as well
 }
 
+/**
+ * This function helps build resolvers from common callback data. Callback data
+ * provided by many functions contains error information in the first argument
+ * and an arbitrary response across all other arguments. It is common to build a
```

Comment:
Would it be possible to make resolver into an object, and replace this with a constructor?